### ISSUE TITLE
SpanReporter does not close cleanly

### ISSUE THREAD

**Issue by AgDude** [wavefrontHQ/wavefront-opentracing-sdk-python#36]
I am expecting that after calling the `close` method the sending thread will be terminated. This is not happening.   If the buffer is already empty the thread will be blocked on the call to `Queue.get`, and never evaluate `self._stop`.  The `thread.join` method does not terminate the thread after the timeout, it simply ignores it. 

So the current close method causes the thread to exit after sending the next span (regardless of the number present in the buffer), since `self._stop` will now be `True`, regardless of the timeout on join. If there are many items in the buffer, only one will be sent. If there are no items in the buffer, the thread will persist until one is added to the buffer and sent.  The `join` thus has no effect, other than blocking the main thread for 10 seconds if the buffer is empty when close is called.

**Comment by sushantdewan123** [comment]
Won't fix this issue for now.
